File: MedicalDonation/src/main/java/com/MedicalDonation/service/PostService.java
package com.MedicalDonation.service;


import com.MedicalDonation.dto.PostRequest;
import com.MedicalDonation.model.Post;
import com.MedicalDonation.repository.PostRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service

public class PostService {
    @Autowired
    public PostRepository postRepository;
    public void save(PostRequest postRequest){
        Post post=new Post();
        post.setId(postRequest.getId());
        post.setTitle(postRequest.getTitle());
        post.setImage(postRequest.getImage());
        post.setMfd_Date(postRequest.getMfd_Date());
        post.setExp_Date(postRequest.getExp_Date());
        post.setQuantity(postRequest.getQuantity());
        postRepository.save(post);

    }
    public Post getPost(Long id){
        return postRepository.findById(id).get();
    }
    public List<Post> getAllPost(){
        return postRepository.findAll();
    }
    public void delete(Long id){
        postRepository.deleteById(id);
    }
    public void update(PostRequest postRequest) {
        Post post = new Post();
        post.setId(postRequest.getId());
        if(postRequest.getTitle()!=null)
        post.setTitle(postRequest.getTitle());
        if(postRequest.getImage()!=null)
        post.setImage(postRequest.getImage());
        if(postRequest.getMfd_Date()!=null)
        post.setMfd_Date(postRequest.getMfd_Date());
        if(postRequest.getExp_Date()!=null)
        post.setExp_Date(postRequest.getExp_Date());
        if(postRequest.getQuantity()!=null)
        post.setQuantity(postRequest.getQuantity());
        postRepository.save(post);
    }
    public void updateQuantity(Long id, Integer quantity) {
        Post post = postRepository.findById(id).get();
        post.setQuantity(quantity);
        postRepository.save(post);
    }
}

File: MedicalDonation/src/app/service/post.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Post } from '../model/post';

@Injectable({
  providedIn: 'root',
})
export class PostService {
  constructor(private http: HttpClient) {}

  save(post: Post) {
    return this.http.post('http://localhost:8084/api/post/create', post);
  }

  update(post: Post) {
    return this.http.put('http://localhost:8084/api/post/update', post);
  }
  updateQuantity(id:number|null,q:number) {
    let data=[id,q];
    return this.http.put(`http://localhost:8084/api/post/quantity/${data[0]}/${data[1]}`,data);
  }
  getPost(id: number|null) {
    return this.http.get(`http://localhost:8084/api/post/${id}`);
  }

  delete(id: number|null) {
    return this.http.delete(`http://localhost:8084/api/post/${id}`);
  }

  getAllPost() {
    return this.http.get('http://localhost:8084/api/post/all');
  }
}